### PR TITLE
ESLint 3.x updates

### PR DESCRIPTION
**Pull request by nschonni**
- Bump to the 3.x branch since we only run it in Node 4
- Add the recommended options, but left our existing overrides
- Rename the eslintrc file to JSON to match the newer format that allows it to skip guessing
- Delete the old jshintignores since it is no longer applicable
- Enforce and fix a few variable declaration issues
- Ensure the lint runs during the regular Travis script section so it fails the build

### CODE REVIEW

File: bin/node-sass [hunk 1]
```diff
@@ -182,6 +182,7 @@ function getEmitter() {
  */
 
 function getOptions(args, options) {
+  var cssDir, sassDir, file, mapDir;
```
Comment (xzyfer) [inline bin/node-sass]: Maybe it's personal preference I find this gross.

Comment (xzyfer) [inline bin/node-sass]: That's probably an argument for cleaning up the cli code in general.

**Comment by xzyfer** [comment]
lgtm

File: bin/node-sass (hunk 1 above)
Comment (nschonni) [inline bin/node-sass]: Yeah, if we get to just versions that support ES2015, then we can use stuff like `let` and `const`, but I don't think it's work adding in something like Babel just to get that right now

**Pull request merged**